src/upkeeps/UpKeepManager.sol

```diff
@@ -0,0 +1,496 @@
+
+    event NewMember(address indexed memberAddress, string name, uint256 gasLimit, uint256 timestamp);
+    event RemoveMember(address indexed memberAddress, uint256 upKeepId, uint256 timestamp);
+
+    event RoundsTopUpUpdated(uint256 oldValue, uint256 newValue);
+    event MinRoundsTopUpUpdated(uint256 oldValue, uint256 newValue);
+
+    event SweepLinkToTechops(uint256 amount, uint256 timestamp);
+    event SweepEth(address recipient, uint256 amount, uint256 timestamp);
+    event EthSwappedForLink(uint256 amountEthOut, uint256 amountLinkIn, uint256 timestamp);
+
+    event RegistryEthReceived(address indexed sender, uint256 value);
+
+    constructor(address _governance) {
+        if (_governance == address(0)) revert ZeroAddress();
+        governance = _governance;
+
+        roundsTopUp = 20;
+        minRoundsTopUp = 3;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // MODIFIERS
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks whether a call is from the governance.
+    modifier onlyGovernance() {
+        if (msg.sender != governance) {
+            revert NotGovernance(msg.sender);
+        }
+        _;
+    }
+
+    /// @notice Checks whether a call is from the keeper.
+    modifier onlyKeeper() {
+        if (msg.sender != address(CL_REGISTRY)) {
+            revert NotKeeper(msg.sender);
+        }
+        _;
+    }
+
+    /// @dev Fallback function accepts Ether transactions.
+    receive() external payable {
+        emit RegistryEthReceived(msg.sender, msg.value);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Governance
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @dev It will initiate the upKeep job for monitoring members
+    /// @notice only callable via governance
+    /// @param _gasLimit gas limit for the upKeepManager monitoring upkeep task
+    function initializeBaseUpkeep(uint256 _gasLimit) external onlyGovernance {
+        if (_gasLimit == 0) revert ZeroUintValue();
+
+        monitoringUpKeepId = _registerUpKeep(address(this), _gasLimit, NAME);
+
+        if (monitoringUpKeepId > 0) {
+            /// @dev give allowance of spending LINK funds
+            LINK.approve(address(CL_REGISTRY), type(uint256).max);
+        }
+    }
+
+    /// @dev Adds an member into the manager
+    /// @notice only callable via governance
+    /// @param _memberAddress contract address to be register as new member
+    /// @param _name member's name
+    /// @param _gasLimit amount of gas to provide the target contract when performing upkeep
+    /// @param _existingUpKeepId optional upKeepId to add member which has being registered from other admins
+    function addMember(address _memberAddress, string memory _name, uint256 _gasLimit, uint256 _existingUpKeepId)
```

Comment:
Very minor
Add check to `_gasLimit` to avoid adding member that can be replaced

Reply:
mind elaborating, which check?

there is checks for null values

```
if (_gasLimit == 0) revert ZeroUintValue();
```

Reply:
My bad LG